### DISCUSSION TITLE
How to acccess shapes in OrchardCore.Menu?

### DISCUSSION THREAD

**Discussion by Eaglef90** [OrchardCMS/OrchardCore#12354] [Q&A] [answered]
I am new to Orchard Core and have what is probibily a stupid question but it is not address in the refences section of the docs, that I can tell at least.

I am using OrchredCore as a decompuled CMS and have gone into the admin site and setup a menu called "Site Navigation" in my View's controller I want to fill a model with the data for that navigation I created but I can't figure out how to use "OrchardCore.Menu" to fill that model so I can start calling the .Items to get the MenuItems data to navigate though and build my own custom navigation. The code samples in the references page shoe me how to use the properties of OrchardCore.Menu but not how to make an me OrchardCore.Menu and fill it with the data I need. How do I do this? Please help, this is the last thing I need to finish the general layout of my site so I can start publishing content.

**Comment by Skrypt** [comment, marked as answer]
Look at our themes they all integrate a menu with corresponding templates.

**Reply by Eaglef90** [reply]
I'm sorry. I am too stupid to figure this out. I have been looking over the themes but I see the same problem. I can't figure out how to fill the model. I looked at a few themes but look at "TheTheme" theme. I see the '@using OrchardCore.Model' in the _ViewImparts file and in the Menu and MenuItem views I just see Model called but no fill in it.  Even the HomeController file does not fill the model. So I am still lost on how to fill this model. 

I  am probibly missing something you pros take as second hand but I can't see it anywhere. I did find a blog post at https://orcharddojo.net/blog/how-to-add-a-breadcrumb-menu-in-orchard-core-orchard-nuggets which I think shows what I need but now I have to learn Direct injection so I am kind of still stuck right now.

**Reply by Skrypt** [reply]
I think you will need to create a menu shape in your Layout.cshtml file.

https://docs.orchardcore.net/en/dev/docs/reference/modules/Menu/

**Reply by Eaglef90** [reply]
That is the thing. I have read that doc you referenced and understand how to access the info once a "shape" is made but it does not tell you how to create the menu shape or popuplate it with the right menu. I have never heard of shapes before either. It also looks like all the docs assume your writting in Razor pages or Liquid but I am writting in MVC

**Reply by Skrypt** [reply]
Liquid: 
```liquid
    {% shape "menu", alias: "alias:main-menu", cache_id: "main-menu", cache_fixed_duration: "00:05:00", cache_tag: "alias:main-menu" %}
```

C#:
```c#
    <menu alias="menu-name" cache-id="main-menu", cache-fixed-duration="00:05:00", cache-tag="alias:main-menu" />
```

The Menu uses a TagHelper in Razor. The MenuTagHelper class inherits from BaseShapeTagHelper. As you can see in Liquid we define a Shape element. Here, in Razor you can use the ShapeTagHelper or MenuTagHelper ... they are equivalent.

These are common TagHelpers that can be found in Orchard Core. We would need to document them.

**Reply by Eaglef90** [reply]
I think that is it. I did see that in the ViewImports file of the "TheTheme" theme therew as a special tag imported  but I did not make the connection. I will give that a try later this week when I have time off from school to try it some more. Thank you
